### PR TITLE
BUG: Fix handling of size=() in Generator.choice when a.ndim > 1.

### LINKED ISSUE #26518: BUG: Generator.choice fails with a.ndim > 1 and size=()
### Describe the issue:

Generator.choice with explict size=() fails whereas it works with size=None, when a.ndim > 1

According to the docs this should be a valid operation.

### Reproduce the code example:

```python
import numpy as np

rng = np.random.default_rng()
rng.choice([[1, 2, 3]], size=())  # ValueError: setting an array element with a sequence.
```

### Error message:

```shell
TypeError: only length-1 arrays can be converted to Python scalars
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/home/ricardo/miniconda3/envs/pytensor/lib/python3.12/site-packages/IPython/core/interactiveshell.py", line 3577, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-3-4a79a97119ec>", line 4, in <module>
    rng.choice(np.array([[1, 2, 3]]), size=()) # ValueError: setting an array element with a sequence.
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "numpy/random/_generator.pyx", line 938, in numpy.random._generator.Generator.choice
ValueError: setting an array element with a sequence.
```

### Python and NumPy Versions:

1.26.4
3.12.2 | packaged by conda-forge | (main, Feb 16 2024, 20:50:58) [GCC 12.3.0]

### Runtime Environment:

_No response_

### Context for the issue:

_No response_

### PR DESCRIPTION
**Pull request by WarrenWeckesser**
Closes gh-26518.

### CODE REVIEW

File: numpy/random/tests/test_generator_mt19937.py [hunk 1]
```diff
@@ -933,6 +933,11 @@ def test_choice_large_sample(self):
         res = hashlib.sha256(actual.view(np.int8)).hexdigest()
         assert_(choice_hash == res)
 
+    def test_choice_array_size_empty_tuple(self):
+        assert_array_equal(random.choice([[1, 2, 3]], size=()), [1, 2, 3])
+        assert_array_equal(random.choice([[1]], size=()), [1])
+        assert_array_equal(random.choice([[1]], size=(), axis=1), [1])
```
Comment (rkern) [inline numpy/random/tests/test_generator_mt19937.py:939]: Just for thoroughness, can you also add tests for `[1, 2, 3]`? I haven't looked in detail, but I'm guessing we don't actually cover that case elsewhere.

**Review by rkern — APPROVED** [review]
Minor request, and pending on decision to ignore the sparse CI failures as (I assume) unrelated.

File: numpy/random/tests/test_generator_mt19937.py (hunk 1 above)
Comment (WarrenWeckesser) [reply numpy/random/tests/test_generator_mt19937.py]: Done.  I also added the strict parameter to a couple of the tests to ensure that an incorrect scalar array doesn't sneak in there.

**Comment by charris** [comment]
Thanks Warren.

**Comment by charris** [comment]
Merges of `main` don't backport, please rebase in the future.

**Pull request merged**